atlasaction/action.go

```diff
@@ -79,6 +79,107 @@ func MigrateApply(ctx context.Context, client *atlasexec.Client, act *githubacti
 	return nil
 }
 
+const (
+	StatePending  = "PENDING_USER"
+	StateApproved = "APPROVED"
+	StateAborted  = "ABORTED"
+	StateApplied  = "APPLIED"
+)
+
+// MigrateDown runs the GitHub Action for "ariga/atlas-action/migrate/down".
+func MigrateDown(ctx context.Context, client *atlasexec.Client, act *githubactions.Action) (err error) {
+	var vars atlasexec.Vars
+	if v := act.GetInput("vars"); v != "" {
+		if err := json.Unmarshal([]byte(v), &vars); err != nil {
+			return fmt.Errorf("failed to parse vars: %w", err)
+		}
+	}
+	var a uint64
+	if i := act.GetInput("amount"); i != "" {
+		a, err = strconv.ParseUint(act.GetInput("amount"), 10, 64)
+		if err != nil {
+			return fmt.Errorf("failed to parse amount: %w", err)
+		}
+	}
+	params := &atlasexec.MigrateDownParams{
+		URL:       act.GetInput("url"),
+		DevURL:    act.GetInput("dev-url"),
+		DirURL:    act.GetInput("dir"),
+		ConfigURL: act.GetInput("config"),
+		Env:       act.GetInput("env"),
+		ToVersion: act.GetInput("to-version"),
+		ToTag:     act.GetInput("to-tag"),
+		Amount:    a,
+		Context: &atlasexec.DeployRunContext{
+			TriggerType:    atlasexec.TriggerTypeGithubAction,
+			TriggerVersion: Version,
+		},
+		Vars: vars,
+	}
+	if a := act.GetInput("amount"); a != "" {
```

Comment:
Why you do it twice?

Reply:
Good question. I guess it's an artifact of a refacto. Good catch